endpoint.php

```diff
@@ -83,14 +92,20 @@ function getRequestMethod() {
 }
 
 function getS3Client() {
-    global $serverPublicKey, $serverPrivateKey;
-
-    return S3Client::factory(array(
-        'key' => $serverPublicKey,
-        'secret' => $serverPrivateKey
-    ));
+    global $serverPublicKey, $serverPrivateKey, $s3Region;
+
+    return new S3Client([
+        'region' => $s3Region,
+        'version' => '2006-03-01',
+        'signature_version' => $s3Signature_version,
```

Comment:
Signature version have "v{number}" format. It will be problem, because your implementation support only digits